buildSrc/src/main/java/net/neoforged/neodev/NeoDevPlugin.java

```diff
@@ -0,0 +1,527 @@
+                },
+                downloadAssets.flatMap(DownloadAssets::getAssetPropertiesFile)
+        );
+        // TODO: Gradle run tasks should be moved to gradle group GROUP
+
+        /*
+         * OTHER TASKS
+         */
+
+        // Generate source patches into a patch archive.
+        var genSourcePatches = tasks.register("generateSourcePatches", GenerateSourcePatches.class, task -> {
+            task.setGroup(INTERNAL_GROUP);
+            task.getOriginalJar().set(applyAt.flatMap(ApplyAccessTransformer::getOutputJar));
+            task.getModifiedSources().set(project.file("src/main/java"));
+            task.getPatchesJar().set(neoDevBuildDir.map(dir -> dir.file("source-patches.zip")));
+        });
+
+        // Update the patch/ folder with the current patches.
+        tasks.register("genPatches", Sync.class, task -> {
+            task.setGroup(GROUP);
+            task.from(project.zipTree(genSourcePatches.flatMap(GenerateSourcePatches::getPatchesJar)));
+            task.into(project.getRootProject().file("patches"));
+        });
+
+        // Universal jar = the jar that contains NeoForge classes
+        // TODO: signing?
+        var universalJar = tasks.register("universalJar", Jar.class, task -> {
+            task.setGroup(INTERNAL_GROUP);
+            task.getArchiveClassifier().set("universal");
+
+            task.from(project.zipTree(
+                    tasks.named("jar", Jar.class).flatMap(AbstractArchiveTask::getArchiveFile)));
+            task.exclude("net/minecraft/**");
+            task.exclude("com/**");
+            task.exclude("mcp/**");
+
+            task.manifest(manifest -> {
+                manifest.attributes(Map.of("FML-System-Mods", "neoforge"));
+                manifest.attributes(
+                        Map.of(
+                                "Specification-Title", "NeoForge",
+                                "Specification-Vendor", "NeoForge",
+                                "Specification-Version", project.getVersion().toString().substring(0, project.getVersion().toString().lastIndexOf(".")),
+                                "Implementation-Title", project.getGroup(),
+                                "Implementation-Version", project.getVersion(),
+                                "Implementation-Vendor", "NeoForged"),
+                        "net/neoforged/neoforge/internal/versions/neoforge/");
+                manifest.attributes(
+                        Map.of(
+                                "Specification-Title", "Minecraft",
+                                "Specification-Vendor", "Mojang",
+                                "Specification-Version", minecraftVersion,
+                                "Implementation-Title", "MCP",
+                                "Implementation-Version", mcAndNeoFormVersion,
+                                "Implementation-Vendor", "NeoForged"),
+                        "net/neoforged/neoforge/versions/neoform/");
+            });
```

Comment:
Maybe you should mention in a comment that these attributes are set to these packages to be able to infer the versions in `NeoForgeVersion` without CLI args. Tho the NeoForm version seems not to be used currently

Reply:
Added a comment. NeoForm version is currently unused but that was already the case before. 🤷